OpenRA.Game/Actor.cs

```diff
@@ -160,6 +160,23 @@ Rectangle DetermineVisualBounds()
 
 		public void Tick()
 		{
+			var activity_iterator = currentActivity;
+			List<Target> currentTargets = new List<Target>();
+			while (activity_iterator != null)
+			{
+				if (activity_iterator.GetDestination(this) != CPos.Null && !activity_iterator.IsCanceled)
+				{
+					currentTargets.Add(Target.FromCell(World, activity_iterator.GetDestination(this)));
+				}
+
+				activity_iterator = activity_iterator.NextActivity;
+			}
+
+			if (currentActivity != null)
+			{
+				currentActivity.DrawLines(this, currentTargets);
+			}
+
```

Comment:
This should probably not happen in Tick and not on the actor itself, for the first thing we probably want it in some render method, I can´t give you a suggestion since I am at work atm.

For the second this is because many of the Actors don't need this like buildings, trees etc so making this a trait that can be put on actors that require it is better.